lang/cs/Org.Apache.REEF.Bridge.Core.Grpc/Driver/DriverServiceClient.cs

```diff
@@ -0,0 +1,266 @@
+            var registration = new DriverClientRegistration
+            {
+                Host = host,
+                Port = port
+            };
+            _driverServiceStub.RegisterDriverClient(registration);
+        }
+
+        public void OnShutdown()
+        {
+            Logger.Log(Level.Info, "Driver clean shutdown");
+            _driverServiceStub.Shutdown(new ShutdownRequest());
+        }
+
+        public void OnShutdown(Exception ex)
+        {
+            Logger.Log(Level.Error, "Driver shutdown with error", ex);
+            byte[] errorBytes;
+            try
+            {
+                errorBytes = ByteUtilities.SerializeToBinaryFormat(ex);
+            }
+            catch (SerializationException se)
+            {
+                Logger.Log(Level.Warning, "Unable to serialize exception", ex);
+                errorBytes = ByteUtilities.SerializeToBinaryFormat(
+                    NonSerializableJobException.UnableToSerialize(ex, se));
+            }
+
+            _driverServiceStub.Shutdown(new ShutdownRequest()
+            {
+                Exception = new ExceptionInfo()
+                {
+                    NoError = false,
+                    Message = ex.Message,
+                    Name = ex.Source,
+                    Data = ByteString.CopyFrom(errorBytes)
+                }
+            });
+        }
+
+        public void OnSetAlarm(string alarmId, long timeoutMs)
+        {
+            _driverServiceStub.SetAlarm(new AlarmRequest()
+            {
+                AlarmId = alarmId,
+                TimeoutMs = (int) timeoutMs
+            });
+        }
+
+        public void OnEvaluatorRequest(IEvaluatorRequest evaluatorRequest)
+        {
+            var request = new ResourceRequest()
+            {
+                ResourceCount = evaluatorRequest.Number,
+                Cores = evaluatorRequest.VirtualCore,
+                MemorySize = evaluatorRequest.MemoryMegaBytes,
+                RelaxLocality = evaluatorRequest.RelaxLocality,
+                RuntimeName = evaluatorRequest.RuntimeName,
+                NodeLabel = evaluatorRequest.NodeLabelExpression
+            };
+            if (!string.Empty.Equals(evaluatorRequest.Rack))
+            {
+                request.RackNameList.Add(evaluatorRequest.Rack);
+            }
+
+            if (evaluatorRequest.NodeNames.Count > 0)
+            {
+                request.NodeNameList.Add(evaluatorRequest.NodeNames);
+            }
+            _driverServiceStub.RequestResources(request);
+        }
+
+        public void OnEvaluatorClose(string evalautorId)
+        {
+            _driverServiceStub.AllocatedEvaluatorOp(new AllocatedEvaluatorRequest()
+            {
+                EvaluatorId = evalautorId,
+                CloseEvaluator = true
+            });
+        }
+
+        public void OnEvaluatorSubmit(
+            string evaluatorId, 
```

Comment:
here and everywhere: trim the trailing whitespaces. use CodeMaid or some other tool to automate the process